### PR TITLE
Remove broken URL for cat.component_templates API

### PR DESCRIPTION
**Pull request by JoshMock**
Updates a broken URL for the `cat.component_templates` API spec. It has a typo, and there is no doc for it, so I linked to the component templates docs instead.

It currently has to be manually removed from the [auto-generated API docs for the JS client](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/api-reference.html#_component_templates) every time we generate them.

### CODE REVIEW

File: rest-api-spec/src/main/resources/rest-api-spec/api/cat.component_templates.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "cat.component_templates":{
     "documentation":{
-      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/master/cat-compoentn-templates.html",
+      "url": "https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-component-template.html",
```
Comment (abdonpijpelink) [inline rest-api-spec/src/main/resources/rest-api-spec/api/cat.component_templates.json:4]:
```suggestion
      "url": "https://www.elastic.co/guide/en/elasticsearch/reference/master/cat-component-templates.html",
```

**Review by abdonpijpelink — COMMENTED** [review]
Once https://github.com/elastic/elasticsearch/pull/95035 is merged, the documentation will be at https://www.elastic.co/guide/en/elasticsearch/reference/master/cat-component-templates.html, if you want to wait for that.

**Comment by JoshMock** [comment]
@abdonpijpelink That's totally fine. Thanks for adding it!

**Comment by abdonpijpelink** [comment]
@JoshMock The PR has been merged and https://www.elastic.co/guide/en/elasticsearch/reference/master/cat-component-templates.html is up

**Review by abdonpijpelink — APPROVED** [review]
LGTM

**Pull request merged**